src/pages/studyView/addChart/AddChart.tsx

```diff
@@ -0,0 +1,238 @@
+import * as React from 'react';
+import {action, computed, observable} from "mobx";
+import {observer} from "mobx-react";
+import styles from "./styles.module.scss";
+import {ChildButton, MainButton, Menu} from 'react-mfb';
+import 'react-mfb/mfb.css';
+import {ChartMeta, ChartMetaDataTypeEnum, NewChart, StudyViewPageStore} from "../StudyViewPageStore";
+import {bind} from 'bind-decorator';
+import classnames from "classnames";
+import * as _ from 'lodash';
+import AddChartByType from "./addChartByType/AddChartByType";
+import {remoteData} from "../../../shared/api/remoteData";
+import CustomCaseSelection from "./customCaseSelection/CustomCaseSelection";
+
+export interface IAddChartProps {
+    store: StudyViewPageStore
+}
+
+enum CurrentOpenedDialogEnum {
+    CUSTOM_GROUPS = 'CUSTOM_GROUPS',
+    ADD_CLINICAL = 'ADD_CLINICAL',
+    ADD_GENOMIC = 'ADD_GENOMIC',
+    CLOSED = 'CLOSED'
+}
+
+type CurrentOpenedDialog = CurrentOpenedDialogEnum.CUSTOM_GROUPS | CurrentOpenedDialogEnum.CLOSED | CurrentOpenedDialogEnum.ADD_GENOMIC | CurrentOpenedDialogEnum.ADD_CLINICAL;
+
+export type ChartOption = {
+    label: string,
+    key: string,
+    disabled?: boolean,
+    selected?: boolean,
+    freq: number
+}
+
+export type AddChartFrequency = { [attrId: string]: number };
+
+@observer
+export default class AddChart extends React.Component<IAddChartProps, {}> {
+    private addCustomGroupsChartTitle = 'Add custom groups as Pie Chart';
+
+    @computed
+    get addGenomicDataChartTitle() {
+        return `Add genomic data chart (${_.filter(this.genomicDataOptions, chartOption => !chartOption.selected).length} more to add)`
+    }
+
+    @computed
+    get addClinicalDataChartTitle() {
+        return `Add clinical data chart (${_.filter(this.clinicalDataOptions, chartOption => !chartOption.selected).length} more to add)`
+    }
+
+    @observable currentOpenedDialog: CurrentOpenedDialog = CurrentOpenedDialogEnum.CLOSED;
+
+    readonly getClinicalDataCount = remoteData<AddChartFrequency>({
+        invoke: async () => {
+            const freq = this.calculateFrequency(_.reduce(this.props.store.clinicalDataWithCount.result, (acc, next, key) => {
+                acc[key] = next;
+                return acc;
+            }, {} as AddChartFrequency));
+            return freq;
+        },
+        default: {}
+    });
+
+
+    readonly getGenomicDataCount = remoteData<AddChartFrequency>({
+        invoke: async () => {
+            const freq = this.calculateFrequency(_.reduce(this.props.store.genomicDataWithCount.result, (acc, next, key) => {
+                acc[key] = next;
+                return acc;
+            }, {} as AddChartFrequency));
+            return freq;
+        },
+        default: {}
+    });
+
+    private calculateFrequency(data: AddChartFrequency): AddChartFrequency {
```

Comment:
we selectedSamples.result length should be passed in.  this function will produce wrong results if it's not loaded.